exporter/kafkaexporter/config.go

```diff
@@ -41,8 +42,47 @@ type Config struct {
 	// Client, and shared by the Producer/Consumer.
 	Metadata Metadata `mapstructure:"metadata"`
 
-	// TODO authentication
-	// TODO batch settings
+	// Authentication defines used authentication mechanism.
+	Authentication Authentication `mapstructure:"authentication"`
+}
+
+// AuthType defines authentication type
+type AuthType string
+
+const (
+	// No authentication
+	AuthTypeNone AuthType = "none"
+	// Plain text authentication
+	AuthTypePlaintext AuthType = "plain_text"
+	// TLS authentication
+	AuthTypeTLS AuthType = "tls"
+	// Kerberos authentication
+	AuthTypeKerberos AuthType = "kerberos"
+)
+
+// Authentication defines authentication.
+type Authentication struct {
+	Type      AuthType                   `mapstructure:"type"`
+	PlainText PlainTextConfig            `mapstructure:"plain_text"`
+	TLS       configtls.TLSClientSetting `mapstructure:"tls"`
+	Kerberos  KerberosConfig             `mapstructure:"kerberos"`
```

Comment:
Use pointers to simulate "union", aka only one is set?

Reply:
union merges structs together it does not set only one.